xarray/tests/test_concat.py

```diff
@@ -1261,3 +1262,90 @@ def test_concat_index_not_same_dim() -> None:
         match=r"Cannot concatenate along dimension 'x' indexes with dimensions.*",
     ):
         concat([ds1, ds2], dim="x")
+
+
+def test_concat_join_coordinate_variables_non_asked_dims():
+    ds1 = Dataset(
+        coords={
+            "x_center": ("x_center", [1, 2, 3]),
+            "x_outer": ("x_outer", [0.5, 1.5, 2.5, 3.5]),
+        },
+    )
+
+    ds2 = Dataset(
+        coords={
+            "x_center": ("x_center", [4, 5, 6]),
+            "x_outer": ("x_outer", [4.5, 5.5, 6.5]),
+        },
+    )
+
+    expected_wrongly_concatenated_xds = Dataset(
+        coords={
+            "x_center": ("x_center", [1, 2, 3, 4, 5, 6]),
+            "x_outer": ("x_outer", [0.5, 1.5, 2.5, 3.5, 4.5, 5.5, 6.5]),
+        },
+    )
+
+    # Using join='outer'
+    # default's broadcast=True will allow the concatenation to surprisingly happen
+    # even if `x_outer` sizes do not match
+    actual_xds = concat(
+        [ds1, ds2],
+        join="outer",
+        dim="x_center",
+        data_vars="different",
+        coords="different",
+    )
+    assert_identical(actual_xds, expected_wrongly_concatenated_xds)
+
+    # Using join='exact'
+    with pytest.raises(
+        ValueError,
+        match=re.escape(
+            "cannot align objects with join='exact' where "
+            "index/labels/sizes are not equal along these coordinates (dimensions): "
+            "'x_outer' ('x_outer',)"
+        ),
+    ):
+        concat(
+            [ds1, ds2],
+            join="exact",
+            dim="x_center",
+            data_vars="different",
+            coords="different",
+        )
+
+
+@pytest.mark.parametrize("join", ("outer", "exact"))
```

Comment:
lets merge this with the previous test.

Reply:
They are not exactly the same, the previous test `test_concat_join_coordinate_variables_non_asked_dims` tests coordinate variables while this one `test_concat_join_non_coordinate_variables` tests non-coordinate variables, also the parametrization is not possible with the previous test as the behaviour differs when using join=outer vs join=exact for coordinate variables